Source/core/JSON.h

```diff
@@ -1937,7 +1937,7 @@ namespace Core {
                             // We are assumed to be opaque, but all quoted string stuff is enclosed between quotes
                             // and should be considered for scope counting.
                             // Check if we are entering or leaving a quoted area in the opaque object
-                            if ((current == '\"') && ((_value.empty() == true) || (_value[_value.length() - 2] != '\\'))) {
```

Comment:
This looks scarry, now also "Test \" escaped quote " triggers a QuotedAreaBit. Pretty sure that there are usescases breaking now as the -2 is added very recently due to issue found in the processing of JSON. It was -1 but than did not find the escaping quote and ceated a mess...

Reply:
If memory serves me right this is one of two that failed: https://github.com/rdkcentral/Thunder/blame/8cc16221e1b5c5550946d6dd212ebb6f8164890a/Tests/unit/core/test_jsonparser.cpp#L1956

Reply:
Contrived example without the modification:

string msg = R"({"params":{"ssid":"iPhone\\"}})";
Core::JSON::String json;
json.FromString(msg);

result 0 : ‘{‘
result 1 : ‘”’, _flagsAndCounters ^= QuotedAreaBit
…
result 8 : ‘”’, _flagsAndCounters ^= QuotedAreaBit
…
result 10 : ‘{‘
result 11 : ‘”’, _flagsAndCounters ^= QuotedAreaBit
…
result 16 : ‘”’, _flagsAndCounters ^= QuotedAreaBit
…
result 18 : ‘”’, _flagsAndCounters ^= QuotedAreaBit
…
result 25 : ‘\’
result 26 : ‘\’
result 27 : ‘”’, No reset of QuotedAreaBit!
result 28 : ‘}’, 1st closing bracket
result 29 : ‘}’, 2nd closing bracket
result 30 : ‘\0’

Deserialise ends with finished equals false, offset equals 32

“Parsing failed: Malformed JSON. Missing closing quotes or brackets”

Reply:
Now the test case for string msg = R"({"params":{"ssid":"iPhone \" and some more text"}})"; will fail :-) PLease add this testcase as well, so added a pull request to comply in both situations..